docs/how-search-works.md

```diff
@@ -0,0 +1,107 @@
+
+Finder-frontend provides the frontend for search. The application accepts
+search queries from users, makes requests to [Search API](https://github.com/alphagov/search-api)
+using those queries, and then renders the results of those queries as HTML.
+
+This guide doesn't go into how Search API handles requests. For more detail
+on this please see the [Search API docs](https://docs.publishing.service.gov.uk/apis/search/search-api.html).
+
+## What happens when I submit a search query?
+
+To render a search page finder-frontend takes a users query parameters such as
+`q=breakfast` and the request path `/breakfast-finder`, and then makes a
+query to two APIs: [Content Store](https://github.com/alphagov/content-store)
+and [Search API](https://github.com/alphagov/search-api).
+
+This is the high level sequence of actions finder-frontend takes when you submit a query:
+
+1. Fetches a content item for the finder
+2. With the content item and a user's query parameters, a Search API query is built
+3. The query is sent to Search API, which responds with search results
+4. The search results and details from the content item are rendered as HTML
+
+### Why do we call the content store?
+
+Pretty much the first thing finder-frontend does when you make a request is
+get hold of a content item.
+
+```ruby
+ContentItem.new('/search/services') # gets content item (from memcached / API)
+```
+
+We call the Content Store to get a content item. Like the majority of pages on GOV.UK, all search pages have an associated [content item](docs/finder-content-item.md).
+
+For example, site search's content item is here: https://www.gov.uk/api/content/search.
+
+The content item plays an important role. The content item determines:
+
+- how we should translate a user's request into a query to Search API
+- how we should render the page (facets available, markup)
+
+#### Example
+
+The [services](https://www.gov.uk/search/services) finder is a search page and
+has [a content item](https://www.gov.uk/api/content/search/services).
+
+The content item helps us to render the page.
+It has fields such as `title` ("Services") and `description` ("Find services
+from government"), which we render on the page.
+
+The content item also affects how we query the Search API.
+
+For example, it provides the field `default_documents_per_page: 20`. This tells us to
+display 20 results per page by default. We'll only fetch the top 20 results
+from Search API in this case.
+
+The `filter` field tells us to apply some filters by default to our Search API
+query. In this case `"content_purpose_supergroup": ["services"]` will be added
+to every search query, but, like `default_documents_per_page`, this will be hidden
+from the user.
+
+There are other more involved bits of information provided by the content item,
+such as facets, which affect both how we render pages and how we query Search API.
+
+Search content items are published by Search API and Specialist Publisher.
+
+### Searching
```

Comment:
The titles are written in different styles:

- "Why do we call the content store?" - is a sentence
- "Searching" - is more like a name (it's not, but I'm really sure how to describe it)

Maybe change this one to "Performing a search" or "Querying Search API" or "Fetching the results" or something